### DISCUSSION TITLE
Custom/edited osm.pbf map to build Valhalla from?

### DISCUSSION THREAD

**Discussion by PhilipNgo** [valhalla/valhalla#3589] [Q&A] [answered]
Hi!

Is it possible to build Valhalla using an edited map that is edited with JOSM and not uploaded to OpenStreetMap?
I currently downloaded a small map and made some edits (added 1 walk path for testing) and downloaded it as a _.osm.pbf_ file.
I then tried to build it with Valhalla docker and I get this error: 
![image](https://user-images.githubusercontent.com/34343658/161726526-bbf9ef0b-d633-4148-9c33-817893689af1.png)

It works with other _pbf_ files as well as the unedited _pbf_ map.
As I understand the node id are unique and created when they are uploaded to OpenStreetMap? 
Not sure if that's the case or not, anyhow, do you have any idea on how to fix this problem?

Thank you!

**Comment by nilsnolde** [comment, marked as answer]
it's entirely possible. that rather sounds like a bug of the josm plugin. osm data has to be sorted, first nodes, then ways, then relations, and within the types in ascending order of ID. looks like the plugin fails to do that.

you can use osmium to sort a pbf: https://docs.osmcode.org/osmium/latest/osmium-sort.html

**Reply by PhilipNgo** [reply]
Hi again!
I tried to sort it with osmium sort but the error seems to still be there.
I tried building Valhalla with a fresh pbf file and that seemed to work.
I then edited that pbf in JOSM by just putting a single node.
I then sorted the file and tried to build but still get the same error.
Any ideas on what might be causing it?

**Reply by PhilipNgo** [reply]
Nevermind! I just needed to use osmium renumber to make sure the max id was correct!
